CONTRIBUTING.md

```diff
@@ -1,32 +1,228 @@
+- [Let us know where you have used What The Hack](#let-us-know-where-you-have-used-what-the-hack)
 
-Privacy information can be found at <https://privacy.microsoft.com/en-us>
+### Before You Start
+
+Before you start contributing and file an issue, make sure you've checked for existing issues:
+   - Before you create a new issue, please do a search in [open issues](https://github.com/microsoft/WhatTheHack/issues) to see if the issue or feature request has already been filed.
+   - If you find your issue already exists, make relevant comments and add your [reaction](https://github.com/blog/2119-add-reaction-to-pull-requests-issues-and-comments). Use a reaction:
+        - 👍 up-vote
+        - 👎 down-vote
+
+## Report a Bug in an Existing Hack
+
+You have found a bug in a hack and want to report it. Great! Something is not right, and it needs to be fixed. 
+
+Please go to the Issues page for the WTH repo and create an ["Issue/Bug"](https://github.com/microsoft/WhatTheHack/issues/new?assignees=&labels=bug&template=bug.yml&title=%5BBug%5D%3A+). 
+
+Let us know which hack has the bug. Is it in the Student guide? Coach guide? Is it a documentation issue? Or an issue with a provided resource file or solution file? The form will guide you on what information you should submit. The WTH team will follow up on your submission. 
+
+We welcome bug fixes!  If you wish fix the bug yourself, please see the section on how to [Contribute an update to an existing hack](#-contribute-an-update-to-an-existing-hack)
+
+## Propose New Hack Topic or Improvement to Existing Hack
+
+You have a proposal on how to improve an existing hack, or you want to suggest a new hack topic. For improving an existing hack, your proposal should go beyond a "bug fix" and be a fair sized improvement or addition of content.
+
+For new hack proposals, consider:
+   - Is this a net new hack topic? Or should your contribution extend or modify an existing hack?
+   - It is okay to have more than one hack on the same technology, but the new hack should be an independent set of challenges that stand on their own.
+
+At this point, you are just giving feedback.  You may, or may not, be interested in authoring or updating content yourself. Thanks in advance for your feedback, we can't wait to read it!
+
+Please go to the Issues page for the WTH repo and create an ["Issue/Proposal"](https://github.com/microsoft/WhatTheHack/issues/new?assignees=&labels=proposal&template=proposal.yml&title=%5BProposal%5D%3A+). 
+
+The form will guide you on what information you should submit. The WTH team will follow up on your proposal. 
+
+We welcome new hacks and improvements to existing hacks. If you intend to 'do the work' you are proposing, read on for how to:
+- [Contribute a new hack to What The Hack](#-contribute-a-new-hack-to-what-the-hack)
+OR
+- [Contribute an update to an existing hack](#-contribute-an-update-to-an-existing-hack)
```

Comment:
I don't think there is supposed to be a dash between the # and the word "contribute", ```#contribute-an-update-to-an-existing-hack```

Reply:
Done